routing_plugin_test.go

```diff
@@ -0,0 +1,537 @@
+						SuccessTime: testTimeMc,
+						SuccessAmt:  1000000,
+					},
+				},
+				// MC state set on the third attempt.
+				{
+					// Discourage Bob - Eugene
+					{
+						NodeFrom: bob,
+						NodeTo:   eugene,
+						FailTime: testTimeMc,
+						FailAmt:  1,
+					},
+					// Discourage Charlie - Eugene
+					{
+						NodeFrom: charlie,
+						NodeTo:   eugene,
+						FailTime: testTimeMc,
+						FailAmt:  1,
+					},
+					// Encourage Dave - Eugene
+					{
+						NodeFrom:    dave,
+						NodeTo:      eugene,
+						SuccessTime: testTimeMc,
+						SuccessAmt:  1000000,
+					},
+				},
+			},
+		},
+		{
+			name: "fork before loop node 3",
+			//
+			//       _____Bob_____
+			//      /             \
+			// Alice               Eugene---Frank---George---Loop
+			//     |\___       ___/                /
+			//     |    Charlie                   /
+			//      \                            /
+			//       \___    ___________________/
+			//           Dave
+			//
+			channels: []testChan{
+				// Alice - Bob
+				{alice, bob, 1, 1000, 1000, 1, 1000, 1},
+				// Alice - Charlie
+				{alice, charlie, 2, 1000, 1000, 1, 1000, 1},
+				// Alice - Dave
+				{alice, dave, 3, 1000, 1000, 1, 1000, 1},
+				// Bob - Eugene
+				{bob, eugene, 4, 1000, 1000, 1, 1000, 1},
+				// Charlie - Eugene
+				{charlie, eugene, 5, 1000, 1000, 2, 1000, 1},
+				// Dave - George (expensive)
+				{dave, george, 6, 1000, 1001, 2, 1000, 1},
+				// Eugene - Frank
+				{eugene, frank, 7, 1000, 1000, 1, 1000, 1},
+				// Frank - George (cheap)
+				{frank, george, 8, 1000, 1000, 1, 1000, 1},
+				// George - Loop
+				{george, loopNode, 9, 1000, 1000, 1, 1000, 1},
+			},
+			initError: nil,
+			missionControlState: [][]lndclient.MissionControlEntry{
+				// The original MC state we start with.
+				{
+					{
+						NodeFrom:    charlie,
+						NodeTo:      eugene,
+						SuccessTime: testTime,
+						SuccessAmt:  10000,
+					},
+				},
+				// MC state set on the second attempt.
+				{
+					// Discourage Frank - George
+					{
+						NodeFrom: frank,
+						NodeTo:   george,
+						FailTime: testTimeMc,
+						FailAmt:  1,
+					},
+					// Encourage Dave - George
+					{
+						NodeFrom:    dave,
+						NodeTo:      george,
+						SuccessTime: testTimeMc,
+						SuccessAmt:  1000000,
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			mockLnd := test.NewMockLnd()
+
+			mockLnd.Channels, mockLnd.ChannelEdges =
+				makeTestNetwork(tc.channels)
+
+			lnd := lndclient.LndServices{
+				Client: mockLnd.Client,
+				Router: mockLnd.Router,
+			}
+
+			testClock := clock.NewTestClock(testTime)
+			plugin := makeRoutingPlugin(
+				RoutingPluginLowHigh, lnd, testClock,
+			)
+			require.NotNil(t, plugin)
+
+			// Set start state for MC.
+			mockLnd.MissionControlState = tc.missionControlState[0]
+
+			// Initialize the routing plugin.
+			require.Equal(
+				t, tc.initError,
+				plugin.Init(
```

Comment:
Do you think it's possible to write tests for the method `Init` only? The current test looks good, though it's more like a functional test than a unit test. Since the `Init` method has quite some complexity in it, maybe dedicated unit tests could help others with understanding the algorithm.